examples/cosign/README.md

```diff
@@ -0,0 +1,162 @@
+# Cosign Pipeline
+
+This pipeline depicts how to build a Golang project from source, package it in a
+container using `ko` and verify that the processed was not tampered with.
+
+## How to run this demo
+
+Execute the following commands from the root of this repository:
+
+```bash
+# Only if a cluster is needed.
+make setup-minikube
+
+# Setup tekton w/ chains
+make setup-tekton-chains tekton-generate-keys
+
+# Run a new pipeline.
+make example-cosign
+# Or re-run the last one.
+# tkn pipeline start ko -L
+
+# Wait until it completes.
+tkn pr logs --last -f
+
+# Ensure it has been signed.
+export TASK_RUN=$(tkn pr describe --last -o json | jq -r '.status.taskRuns | keys[] as $k | {"k": $k, "v": .[$k]} | select(.v.status.taskResults[]?.name | match("IMAGE_URL$")) | .k')
+tkn tr describe "${TASK_RUN}" -o jsonpath='{.metadata.annotations.chains\.tekton\.dev/signed}'
+# Should output "true"
+
+# Export URL of the image created from the pipelinerun as IMAGE_URL.
+export IMAGE_URL=$(tkn pr describe --last -o jsonpath='{..taskResults}' | jq -r '.[] | select(.name | match("IMAGE_URL$")) | .value')
+
+# Double check that the SBOM, the attestation and the signature were uploaded to the OCI.
+crane ls "${IMAGE_URL}"
+# Should output something similar to this:
+#   latest
+#   sha256-7e8eb5bef5ad530a910926df57d73a373ac2860d539eb363c51e0b3479480c88.att
+#   sha256-7e8eb5bef5ad530a910926df57d73a373ac2860d539eb363c51e0b3479480c88.sbom
+#   sha256-7e8eb5bef5ad530a910926df57d73a373ac2860d539eb363c51e0b3479480c88.sig
+```
+
+## Verifications
+
+### SBOM
+
+Based on the [CycloneDX documentation](https://cyclonedx.org/use-cases/),
+there is a lot you can do with it! Let's see in practice how it is.
+
+#### SPDX
+
+`ko` generates an SBOM aside the OCI image. The default format is SPDX, but it
+can be changed to CycloneDX.
+
+Retrieve the SBOM with [cosign]:
+
+```bash
+cosign download sbom "${IMAGE_URL}"
+```
+
+By default it looks like the SBOM does not perform any license detection.
+
+#### CycloneDX
+
+Generate the SBOM with CycloneDX from the source:
+
+```bash
+GOARCH=$(uname -m) GOOS=$(uname -s | tr [:upper:] [:lower:]) \
+  cyclonedx-gomod app \
```

Comment:
Should we include a link to <https://github.com/CycloneDX/cyclonedx-gomod> or `go install github.com/CycloneDX/cyclonedx-gomod/cmd/cyclonedx-gomod@latest` somewhere?

Reply:
Sure, it does not hurt 👍